modules/fbx/data/fbx_mesh_data.cpp

```diff
@@ -382,20 +393,27 @@ MeshInstance *FBXMeshData::create_fbx_mesh(const ImportState &state, const FBXDo
 
 	// Add surfaces.
 	int in_mesh_surface_id = 0;
-	for (const SurfaceId *surface_id = surfaces.next(nullptr); surface_id != nullptr; surface_id = surfaces.next(surface_id)) {
-		SurfaceData *surface = surfaces.getptr(*surface_id);
 
-		surface->surface_tool->generate_tangents();
+	for (OrderedHashMap<SurfaceId, SurfaceData>::Element current_surface = surfaces.front(); current_surface; current_surface = current_surface.next()) {
+		SurfaceData &surface_data = current_surface.value();
+
+		// you can't generate them without a valid uv map.
+		if (surface_data.has_uv_map) {
+			surface_data.surface_tool->generate_tangents();
+		}
 
 		mesh->add_surface_from_arrays(
 				Mesh::PRIMITIVE_TRIANGLES,
-				surface->surface_tool->commit_to_arrays(),
-				surface->morphs,
+				surface_data.surface_tool->commit_to_arrays(),
+				surface_data.morphs,
 				use_compression ? Mesh::ARRAY_COMPRESS_DEFAULT : 0);
 
-		if (surface->material.is_valid()) {
-			mesh->surface_set_name(in_mesh_surface_id, surface->material->get_name());
-			mesh->surface_set_material(in_mesh_surface_id, surface->material);
+		// no material if you don't have uvs
+		if (surface_data.has_uv_map) {
```

Comment:
this is wrong should not be here

TODO: fix bug with max making materials transparent